README.md

```diff
@@ -35,6 +33,38 @@ and usage details please visit: https://github.com/pcengines/pce-fw-builder
 For releases older than v4.0.17 and v4.6.9 use the procedure described in this
 [document](docs/release_process.md)
 
+Branch description
+------------------
+
+* `master` - keeps track of [coreboot's master branch](https://review.coreboot.org/cgit/coreboot.git/log/)
+* `release` - where all releases are merged
+* `develop` - where current development takes place periodically synced with
+  coreboot master
+* `rel_x.y.z` - release branches described below
+
+# Steps for new release
+
+1. Create new branch `rel_x.y.z`, where:
+    * `x` is coreboot major version
+    * `y` is coreboot minor version
+    * `z` is PC Engines firmware fork patch number
+2. Create PR for all required changes for version `rel_x.y.z`
+3. When all changes pushed we closing merge window and start testing - this is
+   typically first week of each month
+4. Perform automated regression testing on `rel_x.y.z`
+5. Fix all required issues and repeat point 4 until fixed - this doesn't mean
+   all tests pass, this mean that approved set passed
+6. If results are accepted merge it to `release` branch
+7. Add tag, which should trigger CI and publish binaries
+8. Merge release branch to develop
+
+# Feature/bug fix development
+
+1. Update `develop` by margining coreboot's master
+2. Checkout feature branch from `develop`
```

Comment:
Feature development isn't a part of release? Why should we checkout feature branch from `develop`?
Somehow it collides with `Steps for new release` section (we merge release branch to develop either).

Reply:
@miczyg1 there are couple conditions:

1. New release branch `rel_x.y.z` always start from `release` branch and merge current `develop`
2. `develop` contain recent coreboot master and features merged
3.  since at each final point of release creating we  merge `release` to `develop`, we assume that if you start the first feature then you get `develop` branch that base don last release, then you merge recent coreboot master and checkout your feature branch, at this point next person may want to create new feature branch so it follows the same steps, get recent coreboot's master merged to develop. When you finish your features `develop` may be far from where you started this situation may be resolved in 2 ways:
    * you merge recent `develop` and test your changes
    * you rebase your feature branch to recent developments and setup PR - this depends on how we treat feature branch as private or public, if public we can only accept the first approach with merging, BTW I think the first approach may be overall better

I'm not sure how it collides with `Steps for new release`

Reply:
@miczyg1 you can read [this](https://nvie.com/posts/a-successful-git-branching-model/) - I want to create something similar but need upstream coexist in this. To answer your concerns:

1. step 2 in is probably wrong since this mostly will mean to merge everything needed to `develop` and merge `develop` to `rel_x.y.z`, but at this point, we moving forward and in step 5 there may be fixes that cam into `rel_x.y.z` because we have to fix something, that's why we have to merge back to `develop` later
2. I should change the order of release and develop workflow since this can be confusing, first you do development on a feature branch, then it is merged to `develop` then at some point we have release.

Is that clear? I will change this document little bit.